contracts/injective-cosmwasm-mock/src/testing/query_spot_market_test.rs

```diff
@@ -0,0 +1,90 @@
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use cosmwasm_std::{Addr, Coin};
+use injective_cosmwasm::{checked_address_to_subaccount_id, MarketId};
+use injective_math::{scale::Scaled, FPDecimal};
+use injective_std::types::injective::exchange::v1beta1::{MsgInstantSpotMarketLaunch, QuerySpotMarketsRequest};
+use injective_test_tube::{injective_cosmwasm::SpotMarketResponse, Account, Exchange, InjectiveTestApp, Module, Wasm};
+
+pub const BASE_DENOM: &str = "inj";
+pub const QUOTE_DENOM: &str = "usdt";
+
+pub fn dec_to_proto(val: FPDecimal) -> String {
+    val.scaled(18).to_string()
+}
+
+#[test]
+#[cfg_attr(not(feature = "integration"), ignore)]
+fn test_instantiation() {
```

Comment:
Instead of one big test this should be one test for each message and query. So a good design might be having something like a `setup()` function that creates a market and balances (and whatever else is required to test all messages).